### ISSUE TITLE
NSString to NSNumber value transformer precision issue

### ISSUE THREAD

**Issue by jbouaziz** [icanzilb/JSONModel#422]
Hey,
There seems to be a cast issue when converting `NSString` to `NSNumber` and in some cases the number changes.
I'm referring to `-(NSNumber*)NSNumberFromNSString:(NSString*)string` in `JSONValueTransfomer.m:180`, :

```
[NSNumber numberWithFloat:[@"16909129" floatValue]]
// outputs 16909128
```

With a slight change, it gets corrected:
```
[NSNumber numberWithDouble:[@"16909129" doubleValue]]
// outputs 16909129
```

Does that make sense?

**Comment by billinghamj** [comment]
Yep! Thanks for flagging this. :)

**Comment by billinghamj** [comment]
Note this is not included in the 1.2.0 release we just made. We'll try to push this out soon.
